docs/user-guide/reference/gpu-annotation.md

```diff
@@ -0,0 +1,191 @@
+## Using GPU Resources in a Numaflow Vertex
+
+### What is a Vertex?
+
+In Numaflow, a **vertex** is a core component of a pipeline. Each vertex represents a processing step, such as:
+- Reading data from a source
+- Transforming data using a user-defined function (UDF)
+- Aggregating or reducing data
+- Writing data to a sink
+
+By connecting multiple vertices, you can build complex data processing pipelines for streaming or batch workloads.
```

Comment:
let's point to vertex.md to explain what vertex is, else the doc will become very verbose.